enumerables.rb

```diff
@@ -0,0 +1,154 @@
+module Enumerable
+  def my_each
+    each do |value|
+      yield(value)
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    each do |value|
+      yield(value, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_array = []
+    each do |value|
+      new_array.push(value) if yield(value)
+    end
+    new_array
+  end
+
+  def my_all?
+    flag = true
+    each do |value|
+      flag = yield(value)
+      break unless flag
+    end
+    flag
+  end
+
+  def my_any?
+    flag = false
+    each do |value|
+      flag = yield(value)
+      break if flag
+    end
+    flag
+  end
+
+  def my_none?
+    flag = false
+    each do |value|
+      flag = yield(value)
+      break if flag
+    end
+    !flag
+  end
+
+  def my_count
+    i = 0
+    if block_given?
+      each do |value|
+        i += 1 if yield(value)
+      end
+    else
+      i = length
+    end
+    i
+  end
+
+  def my_map
+    new_array = []
+    each do |value|
+      new_array.push(yield(value))
+    end
+    new_array
+  end
```

Comment:
* [ ] `my_map` is identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_map returns an Enumerator if no block is given
   ```ruby
    array = [1,5,8,7,9,5,5,55,5]
    p array.my_map #error
    # LocalJumpError:
        no block given (yield)
      # ./enumerables.rb:66:in `block in my_map'
      # ./enumerables.rb:65:in `each'
      # ./enumerables.rb:65:in `my_map'
  ```